### PR TITLE
Typos and basic grammar error fixing

### PR DESCRIPTION
**Pull request by Fourchaux**
Rebase of #1202 

Some discussions #817

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -179,6 +179,10 @@ Working version
 
 - GPR#1187: Minimal documentation for compiler plugins
   (Florian Angeletti)
+  
+- GPR#1202: Fix comments typos as well as basic grammar errors.
```
Comment (mshinwell) [inline Changes]: "comments typos" -> "Typos in comments"

File: bytecomp/lambda.mli [hunk 2]
```diff
@@ -219,7 +219,7 @@ type let_kind = Strict | Alias | StrictOpt | Variable
     Strict: e may have side-effects; always evaluate e first
       (If e is a simple expression, e.g. a variable or constant,
        we may still substitute e'[x/e].)
-    Alias: e is pure, we can substitute e'[x/e] if x has 0 or 1 occurrences
+    Alias: e is pure, we can substitute e'[x/e] if x has 0 or 1 occurrence
```
Comment (mshinwell) [inline bytecomp/lambda.mli]: I think this was correct before.

File: bytecomp/matching.ml [hunk 3]
```diff
@@ -571,7 +571,7 @@ let up_ok (ps,act_p) l =
 
 
 (*
-   Simplify fonction normalize the first column of the match
+   Simplify function normalize the first column of the match
```
Comment (mshinwell) [inline bytecomp/matching.ml]: "Simplify function normalize" -> "The simplify function normalizes"?

File: bytecomp/matching.ml [hunk 4]
```diff
@@ -2813,7 +2813,7 @@ and compile_no_test divide up_ctx repr partial ctx to_match =
    or lazy pattern execute arbitrary code that may perform side effects
    and change the subject values.
 LM:
-   Lazy pattern was PR #5992, initial patch by lwp25.
+   Lazy pattern was PR#5992, initial patch by lwp25.
```
Comment (mshinwell) [inline bytecomp/matching.ml]: This should be lpw25 not lwp25.

File: debugger/time_travel.ml [hunk 5]
```diff
@@ -499,14 +499,14 @@ let rec run () =
   if not !interrupted then
     run ()
 
-(* Run backward the program form current time. *)
+(* Run backward the program from current time. *)
```
Comment (mshinwell) [inline debugger/time_travel.ml]: "Run the program backward"

File: ocamldoc/odoc_ast.ml [hunk 6]
```diff
@@ -677,7 +677,7 @@ module Analyser =
       in
       iter [] [] last_pos (p_cls.Parsetree.pcstr_fields)
 
-    (** Analysis of a [Parsetree.class_expr] and a [Typedtree.class_expr] to get a a couple (class parameters, class kind). *)
+    (** Analysis of a [Parsetree.class_expr] and a [Typedtree.class_expr] to get a couple (class parameters, class kind). *)
```
Comment (mshinwell) [inline ocamldoc/odoc_ast.ml]: A pair, not a couple

File: ocamldoc/odoc_class.ml [hunk 7]
```diff
@@ -43,7 +43,7 @@ and class_apply = {
 
 and class_constr = {
     cco_name : Name.t ; (** The complete name of the applied class *)
-    mutable cco_class : cct option;  (** The associated class ot class type if we found it *)
+    mutable cco_class : cct option;  (** The associated class of class type if we found it *)
```
Comment (mshinwell) [inline ocamldoc/odoc_class.ml]: "of the"

File: ocamldoc/odoc_dep.ml [hunk 8]
```diff
@@ -180,7 +180,7 @@ let type_deps t =
 
   !l
 
-(** Modify the modules depencies of the given list of modules,
+(** Modify the modules dependencies of the given list of modules,
```
Comment (mshinwell) [inline ocamldoc/odoc_dep.ml]: "modules dependencies" -> "module dependencies"

File: ocamldoc/odoc_info.mli [hunk 9]
```diff
@@ -489,7 +489,7 @@ module Module :
         {
           im_name : Name.t ; (** Complete name of the included module. *)
           mutable im_module : mmt option ; (** The included module or module type, if we found it. *)
-          mutable im_info : Odoc_types.info option ; (** comment associated to the includ directive *)
+          mutable im_info : Odoc_types.info option ; (** comment associated to the include directive *)
```
Comment (mshinwell) [inline ocamldoc/odoc_info.mli]: "associated with"

File: ocamldoc/odoc_info.mli [hunk 10]
```diff
@@ -678,7 +678,7 @@ module Module :
 val reset_type_names : unit -> unit
 
 (** [string_of_variance t (covariant, invariant)] returns ["+"] if
-   the given information means "covariant", ["-"] if the it means
+   the given information means "covariant", ["-"] if it means
    "contravariant", orelse [""], and always [""] if the given
    type is not an abstract type with no manifest (i.e. no need
    for the variance to be printed.*)
```
Comment (mshinwell) [inline ocamldoc/odoc_info.mli]: Missing closing parenthesis

File: ocamldoc/odoc_info.mli [hunk 11]
```diff
@@ -1024,7 +1024,7 @@ module Scan :
 (** Computation of dependencies. *)
 module Dep :
   sig
-    (** Modify the modules depencies of the given list of modules,
+    (** Modify the modules dependencies of the given list of modules,
```
Comment (mshinwell) [inline ocamldoc/odoc_info.mli]: "modules dependencies" -> "module dependencies"

File: ocamldoc/odoc_latex.ml [hunk 12]
```diff
@@ -90,7 +90,7 @@ let print_concat fmt sep f =
 (** Generation of LaTeX code from text structures. *)
 class text =
   object (self)
-    (** Return latex code to make a sectionning according to the given level,
+    (** Return latex code to make a sectioning according to the given level,
```
Comment (mshinwell) [inline ocamldoc/odoc_latex.ml]: "section" not "sectioning"

File: ocamldoc/odoc_name.mli [hunk 13]
```diff
@@ -41,7 +41,7 @@ val head : t -> t
 val depth : t -> int
 
 (** Returns true if the first name is a prefix of the second name.
-   If the two names are equals, then if is false (strict prefix).*)
+   If the two names are equals, then it is false (strict prefix).*)
```
Comment (mshinwell) [inline ocamldoc/odoc_name.mli]: "equals" -> "equal"

File: ocamldoc/odoc_parameter.ml [hunk 14]
```diff
@@ -37,7 +37,7 @@ type parameter = param_info
 
 (** Functions *)
 
-(** acces to the name as a string. For tuples, parenthesis and commas are added. *)
+(** access to the name as a string. For tuples, parenthesis and commas are added. *)
```
Comment (mshinwell) [inline ocamldoc/odoc_parameter.ml]: "parentheses" (plural)

File: ocamldoc/odoc_types.mli [hunk 15]
```diff
@@ -56,7 +56,7 @@ and text_element =
   | Superscript of text (** Superscripts. *)
   | Subscript of text (** Subscripts. *)
   | Module_list of string list
-       (** The table of the given modules with their abstract; *)
+       (** The table of the given modules with their abstract. *)
```
Comment (mshinwell) [inline ocamldoc/odoc_types.mli]: "abstracts."

File: ocamldoc/odoc_types.mli [hunk 16]
```diff
@@ -87,7 +87,7 @@ type info = {
     i_sees : see list; (** The list of \@see tags. *)
     i_since : string option; (** The string in the \@since tag. *)
     i_before : (string * text) list; (** the version number and text in \@before tag *)
-    i_deprecated : text option; (** The of the \@deprecated tag. *)
+    i_deprecated : text option; (** The description text of the \@deprecated tag. *)
```
Comment (mshinwell) [inline ocamldoc/odoc_types.mli]: "description text" -> "textual description"

File: stdlib/format.ml [hunk 17]
```diff
@@ -734,7 +734,7 @@ and pp_open_box state indent = pp_open_box_gen state indent Pp_box
 (* Printing queued text.
 
    [pp_print_flush] prints all pending items in the pretty-printer queue and
-   then flushes the the low level output device of the formatter to effectively
+   then flushes the low level output device of the formatter to effectively
```
Comment (mshinwell) [inline stdlib/format.ml]: "effectively" -> "actually"

File: stdlib/scanf.ml [hunk 18]
```diff
@@ -1188,7 +1188,7 @@ let stopper_of_formatting_lit fmting =
 
 
 (******************************************************************************)
-                           (* Readers managment *)
+                           (* Readers management *)
```
Comment (mshinwell) [inline stdlib/scanf.ml]: "Readers" -> "Reader"

File: typing/ctype.ml [hunk 19]
```diff
@@ -1315,7 +1315,7 @@ let subst env level priv abbrev ty params args body =
     raise exn
 
 (*
-   Only the shape of the type matters, not whether is is generic or
+   Only the shape of the type matters, not whether it is generic or
    not. [generic_level] might be somewhat slower, but it ensures
    invariants on types are enforced (decreasing levels.), and we don't
```
Comment (mshinwell) [inline typing/ctype.ml]: Remove "."

File: typing/ctype.mli [hunk 20]
```diff
@@ -228,7 +228,7 @@ val enlarge_type: Env.t -> type_expr -> type_expr * bool
 val subtype: Env.t -> type_expr -> type_expr -> unit -> unit
         (* [subtype env t1 t2] checks that [t1] is a subtype of [t2].
            It accumulates the constraints the type variables must
-           enforce and returns a function that inforce this
+           enforce and returns a function that enforce this
```
Comment (mshinwell) [inline typing/ctype.mli]: "enforce" -> "enforces"

File: typing/subst.mli [hunk 21]
```diff
@@ -21,7 +21,7 @@ type t
 
 (*
    Substitutions are used to translate a type from one context to
-   another.  This requires substituing paths for identifiers, and
+   another.  This requires substituting paths for identifiers, and
    possibly also lowering the level of non-generic variables so that
    it be inferior to the maximum level of the new context.
```
Comment (mshinwell) [inline typing/subst.mli]: "it be" -> "they are"

File: typing/typedecl.ml [hunk 22]
```diff
@@ -933,7 +933,7 @@ let compute_variance_type env check (required, loc) decl tyl =
   (* Prepare *)
   let params = List.map Btype.repr decl.type_params in
   let tvl = ref TypeMap.empty in
-  (* Compute occurences in body *)
+  (* Compute occurrences in body *)
```
Comment (mshinwell) [inline typing/typedecl.ml]: "in body" -> "in the body"

**Review by mshinwell — CHANGES_REQUESTED** [review]

**Comment by mshinwell** [comment]
I've pointed out a few things.  If you could fix those ASAP we can get this merged (today?)

**Comment by Fourchaux** [comment]
@mshinwell 
Applying your suggestions.
Thank you.

**Review by mshinwell — APPROVED** [review]

**Pull request merged**